### PR TITLE
Add translations to tutorials and minicourses

### PR DESCRIPTION
**Pull request by eatyourgreens**
Staging branch URL: https://translations-tutorials.pfe-preview.zooniverse.org/projects/wildcam/wildcam-darien?env=production&language=es

Connects up tutorials and mini-courses to the translations API. If translations are available in the selected language, then step text should use the translation resource.

Uses roughly the same pattern as project and workflow task translations, with a wrapper that decorates its children with translation props. However, the static methods on the tutorial components had to be updated to pass the Redux store, so those components are no longer completely independent of the translations implementation in PFE. That, in turn, might make it difficult to extract them as shared components, so I'm open to suggestions as to how to do that better.

# Required Manual Testing

- [x] Does the non-logged in home page render correctly?
- [x] Does the logged in home page render correctly?
- [x] Does the projects page render correctly?
- [x] Can you load project home pages?
- [x] Can you load the classification page?
- [x] Can you submit a classification?
- [x] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

**Comment by eatyourgreens** [comment]
Missing from this at the moment: there's a bunch of button label text etc. that is coded into the old coffeescript components. That needs to be moved to the locale files so that we can translate the buttons too.

File: app/redux/ducks/translations.js [hunk 1]
```diff
@@ -14,9 +14,9 @@ const initialState = {
   locale: DEFAULT_LOCALE,
   strings: {
     project: {},
-    workflow: {
-      tasks: {}
```
Comment (shaunanoordin) [inline app/redux/ducks/translations.js]: I was initially worried that removing that initial `task: {}` statement here might affect other esoteric parts of the PFE Classifier that blindly calls `workflow.tasks[taskName]` without checking for existence... but, after re-reading the code, I realise that probably isn't happening anywhere yet, and I'm confusing `translations.string.workflow` with _other_ `this.props.workflow` definitions.

Still, reminder to self: test this assumption.

**Comment by shaunanoordin** [comment]
## Quick Thoughts
The code changes themselves LGTM, (barring any code in Coffeescripts, as I can only glance at .cjsx files for a few seconds at a time lest I get cursed,) so I just have to run tests before I can give a full 👍  to this PR.

My only (minor) query: is the [Spanish localisation file](https://github.com/zooniverse/Panoptes-Front-End/blob/afcebc54bd09c64de81370f840a82b247036d0a5/app/locales/es.js) just the English file, but with a few entries missing? Functionally it doesn't matter to the code, but I'm curious to know why we have an es.js file that's not even partially translated into Spanish. (By default, I'm going to assume it's a placeholder for until team Darien provides some translations.)

### Status
Code LGTM, tests are ongoing. Expect 👌  (or 😱 ) by Monday, but ping me if this needs to get done immediately.

File: app/redux/ducks/translations.js (hunk 1 above)
Comment (eatyourgreens) [reply app/redux/ducks/translations.js]: I think this should be ok. `workflow.tasks` was removed in #4199 but I missed this line.

**Comment by eatyourgreens** [comment]
`es.js` is copied directly from the English file, mostly so we can easily see what still needs to be translated. There might be some keys missing, but I've been trying to keep both files updated when the English file changes.

**Comment by shaunanoordin** [comment]
## PR Review
This PR adds translation functionality to Tutorials and Minicourses in the Zooniverse PFE.

Tl;DR: code LGTM, the new functionality is implemented correctly, and basic testing shows that core Zooniverse PFE functionality is retained. However, there seems to be a weird side-effect glitch with Tutorials that I'm trying to confirm.

### Testing Notes
Basic testing looks good. As a logged-in user AND non-logged-in user, viewing the latest version of this branch on `localhost` (with `?env=production` ) on OSX+Chrome63 in Incognito mode, I can...
- view the Zooniverse home page
- log in & log out
- open the Projects page
- select a random project and successfully submit Classifications (random project: Cheetahs of Central Namibia)
- select the primary project being examined and successfully submit Classifications (WildCam Darien)

Primary testing (of the new functionality) also looks good.
- When accessing `//localhost/projects/wildcam/wildcam-darien/?env=production&language=es`, most of the content on the page is translated into Spanish, and **the Tutorial** (that's shown when a user opens the Classifier page) is properly shown in Spanish.
- When accessing the project without the `?language=es` modifier, the project defaults to English as expected.

screen shot 2017-12-11 at 18 11 15
_WildCam Darien showing the Spanish tutorial, powered by the new Translations API._

So, basically, the core functionality of the Zooniverse remains unchanged 👌 and the new changes intended by this PR seem to have been implemented successfully. 👍 Hurrah!

HOWEVER, extended testing did show one really, really freaking bizarre glitch, which will need a separate post to explore WTF is happening. But quite simply:

screen shot 2017-12-11 at 18 16 25
_Scribes of the Cairo Geniza welcomes you to Panama._

### Status
PR looks good save for one side-effect glitch. Investigating and documenting tutorial glitch now.

**Comment by shaunanoordin** [comment]
Update: basic & primary tests look good for OSX+Firefox57/Safari11.

**Comment by eatyourgreens** [comment]
I don't think that's a specific tutorials bug. I think it's a more general bug where the store remains in the previous state if you switch to a project that has no translations. 

Loading a new project should probably reset the store somehow.

**Comment by shaunanoordin** [comment]
## Medium (?) Issue
Once you see the WildCam Darien Tutorial, all further tutorials will always be the WildCam Darien Tutorial.

### Replication Steps
1. Open the WildCam Darien link (https://translations-tutorials.pfe-preview.zooniverse.org/projects/wildcam/wildcam-darien?env=production&language=es) in a new Incognito/Private window, so you'll be a non-logged-in user and always see Tutorials when starting a project.
2. Click "Get Started" to go to WildCam Darien's Classifier page
3. You should see the WildCam Darien Tutorial.
4. In the top Zooniverse navi, click Projects.
5. Select another project, e.g. Scribes of the Cairo Geniza.
6. Click "Get Started" to go to this Cairo G's Classifier page.

Expected: You should see Cairo G's Tutorial.
Actual: You see WildCam Darien's Tutorial.

This bug does not occur if you, e.g. reload the page, or navigate directly to a different project via typing in the URL.

### Testing notes
- This issue has been confirmed on localhost and the testing branch.
- Testing has been confirmed using OSX+Chrome/Firefox
- This issue does _not_ seem to exist on zooniverse.org
  - Test this on: https://www.zooniverse.org/projects/wildcam/wildcam-darien
  - Also test this on: https://www.zooniverse.org/projects/wildcam/wildcam-darien?env=production&language=en and &language=es, to ensure potentially triggering variables are factored.
- I have _not_ tested this on localhost master.

### Analysis
@eatyourgreens I just saw your comments.

> I don't think that's a specific tutorials bug. I think it's a more general bug where the store remains in the previous state if you switch to a project that has no translations.
> Loading a new project should probably reset the store somehow.

Ah, good point! If the problem's anywhere, it's probably in the store - I'll start poking around in there. Loading a new project _doesn't_ seem to reset the store however, which is what's leaving me scratching my head.

@eatyourgreens please let me know if the replication steps above look OK on your machine; if you don't see the same issue as I do, I'm happy to mark this bug as a separate item to investigate/fix. (i.e. _not_ block this PR.)

### Status
Investigation ongoing

**Comment by eatyourgreens** [comment]
I've seen a variant of this issue where a previous project description shows up on the home page after I navigate to a second project, so I think it is caused by old translations remaining in the store after leaving a project.

**Comment by shaunanoordin** [comment]
Ah... ahhh, butts.

I think I was looking at it too narrowly; it's not that the WildCam Darien _tutorial_ is being sticky when navigating to different projects, it's the _whole WildCam Darien project_ being sticky. The tutorials were just a symptom, and the only thing I noticed because I was examining tutorial translations.

1. Go to https://www.zooniverse.org/projects
2. Click on "League of Nations in the Digital Age" in the Projects list.
screen shot 2017-12-11 at 19 13 52
3. Click on the Projects link at the top navi.
4. Click on "WildCam Darien" in the Projects list.
5. Click on the Projects link at the top navi, again.
6. Click on "League of Nations in the Digital Age" in the Projects list, again.
7. SURPRISE the League of Nations is now staffed entirely by large cats.
screen shot 2017-12-11 at 19 14 18

(Strangely, _the Tutorial looks fine on zooniverse.org, but not on the testing branch._ So... anything that's translated is "sticky"? )

You're right @eatyourgreens , this is a general bug, and a hell of a bug it is.
